tasks/tasks.go

```diff
@@ -0,0 +1,194 @@
+package tasks
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/joyent/containerpilot/utils"
+)
+
+// TaskConfig configures tasks that run periodically
+type TaskConfig struct {
+	Name      string   `json:"name"`
+	Args      []string `json:"command"`
+	Frequency string   `json:"frequency"`
+	Timeout   string   `json:"timeout,omitempty"`
+
+	freqDuration    time.Duration
+	timeoutDuration time.Duration
+	cmd             *exec.Cmd
+	quitChannel     chan int
```

Comment:
`Task`/`TaskConfig` has a very different approach to polling than all our other pollable components. Can you walk me thru why we're making this change? I can see that the package API is a little cleaner but it results in the polling logic being repeated (and if we were to refactor the other polling APIs in the same way that multiplies).